README
NAME
    Alt::Log::Any::App::FWR - Log::Any::App which uses
    Log::Dispatch::FileWriteRotate for file output

VERSION
    This document describes version 0.09 of Alt::Log::Any::App::FWR (from
    Perl distribution Alt-Log-Any-App-FWR), released on 2014-05-16.

DESCRIPTION
    This distribution contains an alternate implementation of Log::Any::App.
    This implementation uses a different backend for file output
    (Log::Dispatch::FileWriteRotate instead of Log::Dispatch::FileRotate).
    The goal is to interoperate more easily with other modules that uses
    File::Write::Rotate, like Process::Govern. The parameters are also
    slightly different (no more "DatePattern", adds "period" and "suffix").

    FYI, the future 1.0 of Log::Any::App will probably also use
    Log::Dispatch::FileWriteRotate.

SEE ALSO
    Alt

HOMEPAGE
    Please visit the project's homepage at
    <https://metacpan.org/release/Alt-Log-Any-App-FWR>.

SOURCE
    Source repository is at
    <https://github.com/sharyanto/perl-Alt-Log-Any-App-FWR>.

BUGS
    Please report any bugs or feature requests on the bugtracker website
    <https://rt.cpan.org/Public/Dist/Display.html?Name=Alt-Log-Any-App-FWR>

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

AUTHOR
    Steven Haryanto <stevenharyanto@gmail.com>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Steven Haryanto.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Alt-Log-Any-App-FWR

0.09     2014-05-16 (SHARYANTO)

         - No functional changes.

         - Replace File::Slurp with File::Slurp::Tiny in test suite.


0.08     2014-01-27 (SHARYANTO)

         - Increase Log::Any::App version to 0.44.

         - Add filtering options
           (filter_{text,no_text,citext,no_citext,re,no_re}).


0.07     2013-08-21 (SHARYANTO)

         - Add LOG_ENV.


0.06     2013-01-18 (SHARYANTO)

         - No functional changes. Synchronize Log::Any::App's version with
           Log-Any-App's (0.43).


0.05     2012-12-28 (SHARYANTO)

         - No functional changes. Update Log::Any::App version to sync with
           Log-Any-App's.


0.04     2012-12-28 (SHARYANTO)

         [BUG FIXES]

         - Haven't passed buffer_size to Log::Dispatch::FileWriteRotate.


0.03     2012-12-28 (SHARYANTO)

         - Support buffer_size for file output.


0.02     2012-12-28 (SHARYANTO)

         - No functional changes. Move information about the alternate
           implementation from Log::Any::App to Alt::Log::Any::App::FWR. Fix
           typos.


0.01     2012-12-28 (SHARYANTO)

         - First release as Alt-Log-Any-App-FWR. Renamed from Log-Any-App09.
          Also, version number reset to 0.01 while Log::Any::App is back to 0.41
          (now manually set instead of using # VERSION).


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Log::Any::App which uses Log::Dispatch::FileWriteRotate for file output",
  "dist_author" => [
    "Steven Haryanto <stevenharyanto\@gmail.com>"
  ],
  "dist_name" => "Alt-Log-Any-App-FWR",
  "dist_version" => "0.07",
  "license" => "perl",
  "module_name" => "Alt::Log::Any::App::FWR",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Alt::Base" => "0.01",
    "File::HomeDir" => 0,
    "File::Path" => "2.07",
    "File::Write::Rotate" => "0.10",
    "JSON" => 0,
    "Log::Any" => "0.14",
    "Log::Any::Adapter" => "0.07",
    "Log::Any::Adapter::Log4perl" => "0.06",
    "Log::Dispatch::Dir" => "0.10",
    "Log::Dispatch::FileWriteRotate" => "0.01",
    "Log::Dispatch::Syslog" => "2.29",
    "Log::Log4perl" => "1.36",
    "perl" => "5.008000"
  },
  "script_files" => [],
  "test_requires" => {
    "File::Slurp" => 0
  }
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  my $tr = delete $module_build_args{test_requires};
  my $br = $module_build_args{build_requires};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Alt/Log/Any/App/FWR.pm
package Alt::Log::Any::App::FWR;
use base qw(Alt::Base);

our $VERSION = '0.09'; # VERSION

1;
# ABSTRACT: Log::Any::App which uses Log::Dispatch::FileWriteRotate for file output

__END__

=pod

=encoding UTF-8

=head1 NAME

Alt::Log::Any::App::FWR - Log::Any::App which uses Log::Dispatch::FileWriteRotate for file output

=head1 VERSION

This document describes version 0.09 of Alt::Log::Any::App::FWR (from Perl distribution Alt-Log-Any-App-FWR), released on 2014-05-16.

=head1 DESCRIPTION

This distribution contains an alternate implementation of L<Log::Any::App>. This
implementation uses a different backend for file output
(L<Log::Dispatch::FileWriteRotate> instead of L<Log::Dispatch::FileRotate>). The
goal is to interoperate more easily with other modules that uses
L<File::Write::Rotate>, like L<Process::Govern>. The parameters are also
slightly different (no more C<DatePattern>, adds C<period> and C<suffix>).

FYI, the future 1.0 of Log::Any::App will probably also use
Log::Dispatch::FileWriteRotate.

=head1 SEE ALSO

L<Alt>

=head1 HOMEPAGE

Please visit the project's homepage at L<https://metacpan.org/release/Alt-Log-Any-App-FWR>.

=head1 SOURCE

Source repository is at L<https://github.com/sharyanto/perl-Alt-Log-Any-App-FWR>.

=head1 BUGS

Please report any bugs or feature requests on the bugtracker website L<https://rt.cpan.org/Public/Dist/Display.html?Name=Alt-Log-Any-App-FWR>

When submitting a bug or request, please include a test-file or a
patch to an existing test-file that illustrates the bug or desired
feature.

=head1 AUTHOR

Steven Haryanto <stevenharyanto@gmail.com>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Steven Haryanto.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.015.
use strict;
use warnings;

use 5.008000;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Log::Any::App which uses Log::Dispatch::FileWriteRotate for file output",
  "AUTHOR" => "Steven Haryanto <stevenharyanto\@gmail.com>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Alt-Log-Any-App-FWR",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Alt::Log::Any::App::FWR",
  "PREREQ_PM" => {
    "Alt::Base" => "0.01",
    "Data::Dump" => 0,
    "File::HomeDir" => 0,
    "File::Path" => "2.07",
    "File::Write::Rotate" => "0.10",
    "JSON" => 0,
    "Log::Any" => "0.14",
    "Log::Any::Adapter" => "0.07",
    "Log::Any::Adapter::Log4perl" => "0.06",
    "Log::Dispatch::Dir" => "0.10",
    "Log::Dispatch::FileWriteRotate" => "0.01",
    "Log::Dispatch::Syslog" => "2.29",
    "Log::Log4perl" => "1.36"
  },
  "TEST_REQUIRES" => {
    "File::Slurp::Tiny" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.09",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Alt::Base" => "0.01",
  "Data::Dump" => 0,
  "File::HomeDir" => 0,
  "File::Path" => "2.07",
  "File::Slurp::Tiny" => 0,
  "File::Spec" => 0,
  "File::Write::Rotate" => "0.10",
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "JSON" => 0,
  "Log::Any" => "0.14",
  "Log::Any::Adapter" => "0.07",
  "Log::Any::Adapter::Log4perl" => "0.06",
  "Log::Dispatch::Dir" => "0.10",
  "Log::Dispatch::FileWriteRotate" => "0.01",
  "Log::Dispatch::Syslog" => "2.29",
  "Log::Log4perl" => "1.36",
  "Test::More" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



